aws/src/test/java/nl/nn/adapterframework/filesystem/AmazonS3FileSystemTestHelper.java

```diff
@@ -0,0 +1,203 @@
+package nl.nn.adapterframework.filesystem;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.FilterInputStream;
+import java.io.FilterOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Path;
+import java.util.Iterator;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.AmazonS3Exception;
+import com.amazonaws.services.s3.model.ObjectListing;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+
+import io.findify.s3mock.S3Mock;
+import lombok.Getter;
+import nl.nn.adapterframework.testutil.PropertyUtil;
+import nl.nn.adapterframework.util.StringUtil;
+
+public class AmazonS3FileSystemTestHelper implements IFileSystemTestHelper {
+
+	protected String accessKey    = PropertyUtil.getProperty("AmazonS3.properties", "accessKey");
+	protected String secretKey    = PropertyUtil.getProperty("AmazonS3.properties", "secretKey");
+
+	protected @Getter String bucketName   = PropertyUtil.getProperty("AmazonS3.properties", "bucketName");
+
+	private Regions clientRegion = Regions.EU_WEST_1;
+	public static final int S3_PORT = 19090;
+	private String serviceEndpoint = "http://localhost:"+S3_PORT;
+//	private String serviceEndpoint = null;
```

Comment:
Better to remove this line perhaps?

Is there not a better way to toggle between testing with in-memory mock, and real AWS than static code changes, perhaps based on one of properties?